iocore/eventsystem/P_UnixEThread.h

```diff
@@ -96,11 +96,7 @@ EThread::schedule(Event *e)
   // client VC, it can be HttpCacheSM etc. so save the flags
   e->continuation->control_flags.set_flags(get_cont_flags().get_flags());
 
-  if (e->ethread == this_ethread()) {
-    EventQueueExternal.enqueue_local(e);
-  } else {
-    EventQueueExternal.enqueue(e);
-  }
+  EventQueueExternal.enqueue(e);
```

Comment:
The assumption is `e->ethread == this_ethread()` is always true, because of `e->ethread = this;` in the beginning of this function, right?
Seems `enqueue_local(e)` should be called instead of `enqueue(e)`.

Reply:
I explained the reason for doing this in the linked issue #7327